### PR TITLE
Web app deploy

### PR DESCRIPTION
**Pull request by jdneo**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzureActionsComponent.java [hunk 1]
```diff
@@ -110,6 +110,7 @@ public void initComponent() {
                     actionGroup.addAll((ActionGroup) actionManager.getAction("AzureWebDeployGroup"));
             }
             loadWebAppOnLinux();
+            loadWebApps();
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzureActionsComponent.java]: merge this with loadWebAppOnLinux?

Comment (jdneo) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/AzureActionsComponent.java]: Sure

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppSettingEditor.java [hunk 2]
```diff
@@ -22,31 +22,98 @@
...
+            throw new ConfigurationException("Current project is not a Maven project.");
+        }
+    }
+
+    private boolean shouldAddMavenPackageTask(List<BeforeRunTask> tasks, Project project) {
+        boolean shouldAdd = true;
+        for (BeforeRunTask task: tasks) {
+            if (task.getProviderId().equals(MavenBeforeRunTasksProvider.ID)) {
+                MavenBeforeRunTask mavenTask = (MavenBeforeRunTask) task;
+                if (mavenTask.getGoal().equals(MAVEN_TASK_PACKAGE) &&
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppSettingEditor.java]: contains?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppSettingEditor.java [hunk 3]
```diff
@@ -22,31 +22,98 @@
...
+        String webAppId = mainPanel.getSelectedWebAppId();
+        if (webAppId == null || webAppId.length() == 0) {
+            throw new ConfigurationException("Choose a web app to deploy.");
+        }
+        if (!isMavenProject(project)) {
+            throw new ConfigurationException("Current project is not a Maven project.");
+        }
+    }
+
+    private boolean shouldAddMavenPackageTask(List<BeforeRunTask> tasks, Project project) {
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebAppSettingEditor.java]: this method and isMavenProject should be moved out to a shared utility?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebDeployUtil.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            println(handler,UPLOADING_SUCCESSFUL, ProcessOutputTypes.STDOUT);
+            println(handler,LOGGING_OUT, ProcessOutputTypes.STDOUT);
+            ftp.logout();
+            input.close();
+            if (ftp.isConnected()) {
+                ftp.disconnect();
+            }
+            return null;
+        })
+        .subscribeOn(Schedulers.io())
```
Comment (adashen) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/WebDeployUtil.java]: use the wrappered version

**Comment by jdneo** [comment]
retest this please

**Pull request merged**